src/main/java/org/spongepowered/common/event/tracking/context/transaction/NeighborNotification.java

```diff
@@ -106,11 +130,9 @@ public String toString() {
 
     @Override
     public Optional<BiConsumer<PhaseContext<@NonNull ?>, CauseStackManager.StackFrame>> getFrameMutator(
-        @Nullable GameTransaction<@NonNull ?> parent
+        @Nullable final GameTransaction<@NonNull ?> parent
     ) {
-        return Optional.of((context, frame) -> {
-            frame.pushCause(this.locatableBlock.get());
-        });
+        return Optional.of((context, frame) -> frame.pushCause(this.ticketSupplier.get()));
```

Comment:
This is where I'm debating whether there should be more to the frame mutator, but as it stands, it's suiting nicely
![idea64_MHQmocvSwW.png](https://i.imgur.com/DDyYho9.png)

Reply:
Like, something I could imagine being useful is:
```java
        return Optional.of((context, frame) -> {
            if (parent instanceof ChangeBlock) {
                frame.pushCause(((ChangeBlock) parent).original);
            }
            frame.pushCause(this.ticketSupplier.get());
        });
``` to populate the original changeblock being changed (may look at potentially providing the parent transaction as a BlockTransaction accessor if available)